### PR TITLE
[Fluid] Use materials.json

### PR DESCRIPTION
**Pull request by rubenzorrilla**
Following the Kratos general guidelines and towards the inclusion of the FluidDynamicsApplication in the core applications repository, this PR upgrades the entire application to make the materials.json file mandatory. Note that we have supported both the new and the old material inputs for many months. Take into consideration that only the FluidDynamicsApplication has been upgraded (I considered that making separated PRs for derived applications would simplify the review).

It is due mentioning that this breaks any existent test case (it is only needed to create a materials.json file to make it work again).

Finally I'd like to also add that a temporary solution has been implemented in the two fluids solvers to circumvent the need of the subproperties. This will be removed as soon as these are finally merged to the master (#2919).

I will not merge this until the FluidDynamicsApplication GUI is upgraded accordingly.

### CODE REVIEW

**Comment by philbucher** [comment]
Very nice!!!
What is the default for the file name?

**Comment by rubenzorrilla** [comment]
> Very nice!!!
> What is the default for the file name?

You mean the default 'materials_filename'? If yes, I set it to an empty string to check it is provided. Makes sense for you?

**Comment by loumalouomega** [comment]
> > Very nice!!!
> > What is the default for the file name?
> 
> You mean the default 'materials_filename'? If yes, I set it to an empty string to check it is provided. Makes sense for you?

Default structural is StructuralMaterials.json, should be something like FluidMaterials.json

**Comment by rubenzorrilla** [comment]
> > > Very nice!!!
> > > What is the default for the file name?
> > 
> > 
> > You mean the default 'materials_filename'? If yes, I set it to an empty string to check it is provided. Makes sense for you?
> 
> Default structural is StructuralMaterials.json, should be something like FluidMaterials.json

Done!

**Comment by philbucher** [comment]
> > > Very nice!!!
> > > What is the default for the file name?
> > 
> > 
> > You mean the default 'materials_filename'? If yes, I set it to an empty string to check it is provided. Makes sense for you?
> 
> Default structural is StructuralMaterials.json, should be something like FluidMaterials.json

someone read my mind :)

**Comment by philbucher** [comment]
@rubenzorrilla @KratosMultiphysics/fluid-dynamics some comments from my side, since in StructuralMechanics we use this for a long time already and I collected some experiences:
- I am not sure if dropping the support for properties to be read from mdpa completely is the right way to go. This breaks every single existing case!!! From what I see this PR FIRST introduces the usage of the materials file, @rubenzorrilla it does not seem to me like both versions were available for some transition time...?
- I would give the users some transition time, in which actually both ways are supported.
- StructuralMechanics supports both ways: https://github.com/KratosMultiphysics/Kratos/blob/cc543fda77fc9ce8f57903fc5338c0db6410aa32/applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py#L293-L302
    Personally I find this the best solution, for two reasons: If someone really wants, they can still use the mdpa for the specification of the properties (Note that GiD always writes the Materials file) and second, one can customize how the materials are being read. I myself have for my cases a custom reading-routine, which makes it really well suited to be customized. => Note that the `AnalysisStage` has the function `ModifyInitialProperties` which is designed for exactly this

Long story short, after having struggled a lot with this I think that the solution we have in StructuralMechanics is the most flexible and most suitable for our needs.
=> I recommed Fluid to adapt the same, I don't think it is necessary to have a different solution
If you want I can refactor what we have in StructuralMechanics to the recently added [solver-utils in the core](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/python_scripts/auxiliary_solver_utilities.py), then you could directly use it

**Comment by philbucher** [comment]
Just to add, the Interface would ofc always write the `FluidMaterials.json`, this will be the standard-way (but not the only one)

**Comment by rubenzorrilla** [comment]
My point was that we have supported both import mechanisms for a while (before this PR). I thought that the Kratos general guideline was to only support the json based material import to deprecate the old one. If this is not the case, it shouldn't be too difficult to support both options (with the exception of the two fluids solver, since it will rely on the subproperties as soon as they are merged). 

I'm aware that old cases will not work anymore. Unfortunately, if we decide to only support this material import mechanism there is no easy way to make them compatible. 

Long story short, I completely agree with you if we decide to keep the old mechanism alive. If not, there's nothing else to discuss. Having said this, I think we should discuss **which is (or are) the preferred import mechanisms of core applications** in the @KratosMultiphysics/technical-committee .

**Comment by pooyan-dadvand** [comment]
@KratosMultiphysics/technical-committee agrees on dropping the support for mdpa read properties in solvers for the next release cycle of the Kratos. 

Meanwhile, it was discussed to let users specify an empty string for material.json filename in the ProjectParameters.json which skip the material import giving a warning but not error.

**Comment by rubenzorrilla** [comment]
As we agreed, now we throw a warning if there are no `material_import_settings`. This required to change the default `materials_filename` from `FluidMaterials.json` to an empty string in order to do the check.

So if the `material_import_settings` are not provided the defaults of the solver will set an empty string which triggers the warning. In the GUIs the materials json will always be written as `FluidMaterials.json`.

File: applications/FluidDynamicsApplication/python_scripts/adjoint_fluid_solver.py [hunk 1]
```diff
@@ -54,6 +54,10 @@ def ImportModelPart(self):
 
     def PrepareModelPart(self):
         if not self.main_model_part.ProcessInfo[KratosMultiphysics.IS_RESTARTED]:
+            ## Set fluid properties from materials json file
+            materials_imported = self._SetPhysicalProperties()
+            if not materials_imported:
+                Kratos.Logger.PrintWarning(self.__class__.__name__, "Material properties have not been imported. Check \'material_import_settings\' in your ProjectParameters.json.")
```
Comment (philbucher) [inline applications/FluidDynamicsApplication/python_scripts/adjoint_fluid_solver.py]:
```suggestion
                KratosMultiphysics.Logger.PrintWarning(self.__class__.__name__, "Material properties have not been imported. Check \'material_import_settings\' in your ProjectParameters.json.")
```

**Review by philbucher — DISMISSED** [review]
as agreed on in @KratosMultiphysics/technical-committee

**Comment by philbucher** [comment]
@rubenzorrilla I fixed a copy paste error now travis should compile

**Comment by rubenzorrilla** [comment]
#5591 merging introduced some conflicts. @philbucher can you approve again?

**Review by philbucher — DISMISSED** [review]

**Review by philbucher — APPROVED** [review]

**Comment by philbucher** [comment]
@KratosMultiphysics/potential-flow-team this broke some of your tests can you please have a look?
